python/dex/__init__.py

```diff
@@ -52,15 +52,35 @@ def eval(expr: str, module=prelude, _env=None):
   result = api.evalExpr(_env, api.as_cstr(expr))
   if not result:
     api.raise_from_dex()
-  return Atom(result, module)
+  return Atom._from_ptr(result, module)
 
 
 class Atom:
   __slots__ = ('__weakref__', '_as_parameter_', 'module')
 
-  def __init__(self, ptr, module):
+  def __init__(self, value):
+    catom = api.CAtom()
+    if isinstance(value, int):
+      catom.tag = 0
+      catom.value.tag = 1
+      catom.value.value = ctypes.c_int(value)
+    elif isinstance(value, float):
+      catom.tag = 0
+      catom.value.tag = 4
+      catom.value.value = ctypes.c_float(value)
```

Comment:
Why only `Int64` and `Float32`?

I thought floats in python were `Float64`s precision.
And `Int`'s were arbitrary precision?

I have no idea how ctypes handles these though

Reply:
Just because I wrote this out quickly to unblock you and this was good enough for a simple test. I decided to downcast `int` and `float`, because `Int` and `Float` in Dex are 32-bit. If someone's serious about precision they can always use NumPy scalars to bundle the dtype they want to get on the Dex side.